ocaml/idl/datamodel_pool.ml

```diff
@@ -763,6 +748,19 @@ open Datamodel_types
       ~allowed_roles:_R_POOL_OP
       ()
 
+  let check_update_readiness = call
+      ~name:"check_update_readiness"
+      ~lifecycle:[Published, rel_next, ""]
+      ~doc:"Check if the pool is ready to be updated. If not, report the reasons."
+      ~params:[
+        Ref _pool, "self", "The pool";
+        Bool, "requires_reboot", "Assume that the update will require host reboots"
+      ]
+      ~result:(Set(Set (String)), "A set of error codes with arguments, if the pool is
```

Comment:
as far as I can tell, it should be `Set (Pair String))`? not sure if the datamodel supports that though

Reply:
No, it is not a pair, but list: the first element is always the error code and then there are zero or more arguments (as in datamodel_errors.ml). A better type would be `string * string list`, but the datamodel doesn't support tuples.